CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+
+### Make sure your code is clean
+
+Checkout the [project style guide](TO DO - insert coding convention guidelines), make sure your code is conformant and clean. Remove any debugging lines (`debuggers`, `console.log`).
+
+### Make sure you unit test your changes
+
+Adding a feature? Make sure you add unit tests to support it.
+
+Fixing a bug? Make sure you added a test reproducing the issue.
+
+### Make sure tests pass
+
+All our projects' unit tests can be run by typing `yarn test` at the root of the project. You may need to install dependencies like `mocha`, `chai`, or `sinon`.
+
+### Keep your commit history short and clean
+
+In a large project, it is important to keep the git history clean and tidy. This helps to identify the causes of bugs and helps in identifying the best fixes.
+
+Keeping the history clean means making one commit per feature. It also means squashing every fix you make on your branch after team review.
+
+### Be descriptive
+
+Write a convincing description of your PR and why we should land it.
+
+### Code review
+
+It is important for us to keep the core code clean and consistent. Please ensure any pull request have been reviewed by at least one of the team members.
+
+### Steps for pull request
+
+Above steps should be used when following the steps belowwhen you'd like your work considered for inclusion in the
+project:
+
+1. [Fork](http://help.github.com/fork-a-repo/) the project, clone your fork,
+   and configure the remotes:
+
+   ```bash
+   # Clone your fork of the repo into the current directory
+   git clone https://github.com/<your-username>/<repo-name>
+   # Navigate to the newly cloned directory
+   cd <repo-name>
+   # Assign the original repo to a remote called "upstream"
+   git remote add upstream https://github.com/<upstream-owner>/<repo-name>
+   ```
+
+2. If you cloned a while ago, get the latest changes from upstream:
+
+   ```bash
+   git checkout <dev-branch>
+   git pull upstream <dev-branch>
+   ```
+
+3. Create a new topic branch (off the main project development branch) to
+   contain your feature, change, or fix:
+
+   ```bash
+   git checkout -b <feature/[id]-some-feature>
+   OR
+   git checkout -b <fix/[id]-some-bug>
+   ```
+
+4. Commit your changes in logical chunks. Please adhere to these [git commit
+   message guidelines](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
+   or your code is unlikely be merged into the main project. Use Git's
+   [interactive rebase](https://help.github.com/articles/interactive-rebase)
+   feature to tidy up your commits before making them public.
+
+5. Locally merge (or rebase) the upstream development branch into your topic branch:
+
+   ```bash
+   git pull [--rebase] upstream <dev-branch>
+   ```
+
+6. Push your topic branch up to your fork:
+
+   ```bash
+   git push origin <feature/[id]-some-feature>
+   OR
+   git push origin <fix/[id]-some-bug>
```

Comment:
I think we should go with the convention bug/#id-some-bug

Reply:
Have updated the naming conventions :) Please review.